orchestrator/cli/migrate_domain_models.py

```diff
@@ -0,0 +1,347 @@
+    products: Dict[str, Type[SubscriptionModel]],
+    product_blocks: Dict[str, Type[ProductBlockModel]],
+    db_product_names: List[str],
+    inputs: Dict[str, Dict[str, str]],
+) -> DomainModelChanges:
+    """Map changes that need to be made to fix differences between models and database.
+
+    Args:
+        - model_diffs: Dict with product and product block differences.
+            - products: Dict with product differences.
+            - blocks: Dict with product block differences.
+        - products: Dict with product model by product name.
+        - product_blocks: Dict with product block model by product name.
+        - db_product_names: Product names out of the database.
+        - inputs: Optional Dict with prefilled values.
+
+    Returns: Mapped changes.
+    """
+    create_products = {name: model for name, model in products.items() if name not in db_product_names}
+    delete_products = [name for name in db_product_names if name not in SUBSCRIPTION_MODEL_REGISTRY.keys()]
+
+    # updates need to go before create or deletes.
+    update_product_fixed_inputs = map_update_fixed_inputs(model_diffs["products"])
+    update_resource_types = map_update_resource_types(model_diffs["blocks"], product_blocks, inputs)
+
+    create_resource_type_relations = map_create_resource_type_relations(model_diffs["blocks"])
+    delete_resource_type_relations = map_delete_resource_type_relations(model_diffs["blocks"])
+
+    changes = DomainModelChanges(
+        create_products=create_products,
+        delete_products=delete_products,
+        create_product_fixed_inputs=map_create_fixed_inputs(model_diffs["products"]),
+        update_product_fixed_inputs=update_product_fixed_inputs,
+        delete_product_fixed_inputs=map_delete_fixed_inputs(model_diffs["products"]),
+        create_product_to_block_relations=map_create_product_block_relations(model_diffs["products"]),
+        delete_product_to_block_relations=map_delete_product_block_relations(model_diffs["products"]),
+        update_resource_types=update_resource_types,
+        delete_resource_types=map_delete_resource_types(
+            delete_resource_type_relations, list(update_resource_types.keys()), product_blocks
+        ),
+        create_resource_type_relations=create_resource_type_relations,
+        delete_resource_type_relations=delete_resource_type_relations,
+        create_product_blocks=map_create_product_blocks(product_blocks),
+        delete_product_blocks=map_delete_product_blocks(product_blocks),
+        create_product_block_relations=map_create_product_block_relations(model_diffs["blocks"]),
+        delete_product_block_relations=map_delete_product_block_relations(model_diffs["blocks"]),
+    )
+
+    changes = map_product_additional_relations(changes)
+    changes = map_product_block_additional_relations(changes)
+    changes.create_resource_types = map_create_resource_types(
+        changes.create_resource_type_relations, update_resource_types
+    )
```

Comment:
Think this can also be done in-line just like map_delete_resource_types?

Reply:
no, since we need the `create_resource_type_relations` after it has gone through `map_product_block_additional_relations`.